libs/MeshKernel/include/MeshKernel/TriangulationWrapper.hpp

```diff
@@ -83,117 +73,165 @@ namespace meshkernel
-                faceCounter++;
-                m_faceNodes[f][2] = static_cast<size_t>(faceNodesFlat[faceCounter] - 1);
-                m_faceEdges[f][2] = static_cast<size_t>(faceEdgesFlat[faceCounter] - 1);
-                faceCounter++;
-            }
+        /// @brief Gets the number of triangulated nodes
+        /// @return The number of triangulated nodes
+        [[nodiscard]] const auto& GetNumNodes() const
+        {
+            return m_numNodes;
+        }
 
-            // Create edges
-            if (m_numEdges == 0)
-            {
-                return;
-            }
+        /// @brief Gets the number of triangulated faces
+        /// @return The number of triangulated faces
+        [[nodiscard]] const auto& GetNumFaces() const
+        {
+            return m_numFaces;
+        }
 
-            ResizeAndFill2DVector(m_edgeNodes, m_numEdges, 2, true, constants::missing::sizetValue);
-            size_t edgeCounter = 0;
-            for (size_t e = 0; e < m_numEdges; ++e)
-            {
-                m_edgeNodes[e][0] = static_cast<size_t>(edgeNodesFlat[edgeCounter] - 1);
-                edgeCounter++;
-                m_edgeNodes[e][1] = static_cast<size_t>(edgeNodesFlat[edgeCounter] - 1);
-                edgeCounter++;
-            }
+        /// @brief Gets the triangulated nodes
+        /// @return The triangulated nodes
+        [[nodiscard]] const auto& GetNodes() const
+        {
+            return m_nodes;
+        }
 
-            ResizeAndFill2DVector(m_edgesFaces, m_numEdges, 2, true, constants::missing::sizetValue);
-            edgeCounter = 0;
-            for (size_t f = 0; f < m_numFaces; ++f)
-            {
+        /// @brief Gets the nodes of a triangulated face
+        /// @param faceIndex The face index
+        /// @return The triangulated nodes
+        [[nodiscard]] const auto& GetFaceNodes(const size_t& faceIndex) const
+        {
+            return m_faceNodes[faceIndex];
+        }
 
-                for (size_t n = 0; n < Mesh::m_numNodesInTriangle; ++n)
-                {
-                    auto const edge = static_cast<size_t>(faceEdgesFlat[edgeCounter] - 1);
-                    edgeCounter++;
-                    // For each edge, the shared face index
-                    if (m_edgesFaces[edge][0] == constants::missing::sizetValue)
-                    {
-                        m_edgesFaces[edge][0] = f;
-                    }
-                    else
-                    {
-                        m_edgesFaces[edge][1] = f;
-                    }
-                }
-            }
+        /// @brief Retrieves the face node
+        /// @param faceIndex The index of the face to retrieve the node from
+        /// @param nodeIndex The index of the node to retrieve
+        /// @return const reference to the node with the specified index for the specified face
+        [[nodiscard]] const auto& GetFaceNode(const size_t& faceIndex, const size_t& nodeIndex) const
```

Comment:
Can drop the constant reference in the return type, it's a size_t.
Pass indices by value, no need for constant reference either.